packages/@ember/application/lib/application.ts

```diff
@@ -220,6 +220,16 @@ class Application extends Engine {
     return registry;
   }
 
+  static initializer = buildInitializerMethod<'initializers', Application>(
+    'initializers',
+    'initializer'
+  );
+
+  static instanceInitializer = buildInitializerMethod<'instanceInitializers', ApplicationInstance>(
+    'instanceInitializers',
+    'instance initializer'
+  );
```

Comment:
So this works except for one significant problem: I think we want initializers on Application to return Application/ApplicationInstance not Engine/EngineInstance.

Reply:
Gotcha. Hmm. 🤔 That’s… actually impossible to guarantee from a types POV given the existing API, right?

Reply:
For posterity: We spent some time poking at this and it is indeed impossible to guarantee the correct behavior here… because, and only because, it requires inference on the `this` type, and that interacts badly with `Object.prototype.call`. 😩 Net, we'll leave it as is and… hope for a day when initializers are dead?